### PR TITLE
Add mutex to the cache wrapper

### PR DESCRIPTION
**Pull request by okulik**
We are planning to use determinator cache wrapper from the multiple threads. Even though all the used cache stores are individually thread-safe, Determinator::Cache::FetchWrapper class isn't.

### CODE REVIEW

File: lib/determinator/cache/fetch_wrapper.rb [hunk 1]
```diff
@@ -46,16 +53,20 @@ def key(feature_name)
       # @param url [String] a feature name
       # @return [Feature, MissingResponse] nil when no value is found
       def read_and_upfill(feature_name)
-        @caches.each.with_index do |cache, index|
-          if cache.exist?(key(feature_name))
-            value = cache.read(key(feature_name))
-            @caches[0...index].each do |cache|
-              cache.write(key(feature_name), value)
+        @mutex.synchronize do
```
Comment (me) [inline lib/determinator/cache/fetch_wrapper.rb:56]: Nitpick - this is a private method, might be more readable to have the mutex lock in the caller?

File: lib/determinator/actor_control.rb [hunk 2]
```diff
@@ -24,12 +24,13 @@ def which_variant(name, properties: {})
     end
 
     # @see Determinator::Control#feature_flag_on?
-    def feature_flag_on?(name, properties: {})
+    def feature_flag_on?(name, properties: {}, feature: nil)
```
Comment (me) [inline lib/determinator/actor_control.rb:27]: Should we add this to `which_variant` as well?

File: lib/determinator/cache/fetch_wrapper.rb (hunk 1 above)
Comment (okulik) [reply lib/determinator/cache/fetch_wrapper.rb:56]: Sure thing

**Review by me — APPROVED** [review]
LGTM

**Pull request merged**